internal/app/auth_server.go

```diff
@@ -0,0 +1,226 @@
+package app
+
+import (
+	"context"
+	"crypto/sha256"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"net/url"
+	"strings"
+
+	"github.com/caos/oidc/pkg/op"
+	"github.com/gorilla/mux"
+	"github.com/labstack/echo/v4"
+	"github.com/reearth/reearth-backend/internal/app/appauth"
+	"github.com/reearth/reearth-backend/internal/usecase/interactor"
+	"github.com/reearth/reearth-backend/internal/usecase/interfaces"
+
+	"github.com/golang/gddo/httputil/header"
+)
+
+func authEndPoints(ctx context.Context, e *echo.Echo, r *echo.Group, cfg *ServerConfig) {
+
+	userUsecase := interactor.NewUser(cfg.Repos, cfg.Gateways, cfg.Config.SignupSecret)
+
+	domain, err := url.Parse(cfg.Config.Auth.Domain)
+	if err != nil {
+		panic("not valid auth domain")
+	}
+	domain.Path = "/"
+
+	config := &op.Config{
+		Issuer:                domain.String(),
+		CryptoKey:             sha256.Sum256([]byte(cfg.Config.Auth.Key)),
+		GrantTypeRefreshToken: true,
+	}
+
+	var dn *appauth.AuthDNConfig = nil
+	if cfg.Config.Auth.DN != nil {
+		dn = &appauth.AuthDNConfig{
+			CommonName:         cfg.Config.Auth.DN.CN,
+			Organization:       cfg.Config.Auth.DN.O,
+			OrganizationalUnit: cfg.Config.Auth.DN.OU,
+			Country:            cfg.Config.Auth.DN.C,
+			Locality:           cfg.Config.Auth.DN.L,
+			Province:           cfg.Config.Auth.DN.ST,
+			StreetAddress:      cfg.Config.Auth.DN.Street,
+			PostalCode:         cfg.Config.Auth.DN.PostalCode,
+		}
+	}
+
+	storage := appauth.NewAuthStorage(
+		&appauth.StorageConfig{
+			Domain: domain.String(),
+			Debug:  cfg.Debug,
+			DN:     dn,
+		},
+		userUsecase.GetUserBySubject,
```

Comment:
```suggestion
		func (ctx context.Contrxt, subject string) (*user.User, error) {
  return userUsecase.GetUserBySubject(ctx, subject, getOperator(ctx))
},
```

Operator is required

Reply:
since our end points here are not authenticated, how to use the operator?
the only exception is `userinfo` endpoint and here the `caos/oidc` lib validates the access token.

Reply:
Ok right, operator can be nil. But the usecase func accepts operator as third arg, so it is good to wraps it with new func to clearly show operator is nil, or delete the third arg from the usecase func if operator is not used in the func.

Reply:
I've already removed the operator from the usecase func, that's because its not used